elastica/_rotations.py

```diff
@@ -146,9 +146,13 @@ def _inv_rotate(director_collection):
             )
         )
 
-        # TODO HARDCODED bugfix has to be changed. Remove 1e-14 tolerance
-        theta = arccos(0.5 * trace - 0.5 - 1e-10)
+        # Clip the trace to between -1 and 3.
+        # Any deviation beyond this is numerical error
+        trace = min(trace, 3.0)
+        trace = max(trace, -1.0)
+        theta = arccos(0.5 * trace - 0.5)
```

Comment:
do we have some quantification for our reference regarding the performance penalty this change creates?